enterprise_gateway/services/processproxies/processproxy.py

```diff
@@ -911,7 +911,18 @@ def __init__(self, kernel_manager, proxy_config):
 
     async def launch_process(self, kernel_cmd, **kwargs):
         await super(LocalProcessProxy, self).launch_process(kernel_cmd, **kwargs)
-
+        user=kwargs["env"].get("KERNEL_USERNAME",None)
+        self.log.info(f"KARGS {kwargs}")
+                      
+        if user:
```

Comment:
Can check here if EG_IMPERSONATION_ENABLED & user. (The current EG behavior ignores the KERNEL_USERNAME)

btw,I will also remove unnecessary logging, it was to help me understand how EG works.